### DISCUSSION TITLE
Deploy and build of Eleventy project on Github Pages.

### DISCUSSION THREAD

**Discussion by Kyandroo** [community/community#66523] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hi everyone!

I already saw some information when searching for an answer, but I cannot break my head over it so that's why I'm asking it here. I'm a student and for my internship I'm making a personnal portfolio using Eleventy templating. Everything worked well on my localhost server, but when I tried to deploy it, on first try it didn't want to build at all. The only thing I got in my mailbox was a message: 'pages build and deployment: Some jobs were not successful'.

After some research, I tried putting an .nojekyll file in my root, the result: a 404 not found. So I went searching for another solution. After some time, I found something that you can make a dummy index.html in the root and link to the build folder. Now my html is working, but css styling, images, links and js not. I think it's about the path to the files, so I fixed like one image and that worked, but I have no idea how to fix the links to js or navigation for example. (I'm desperate and would really appreciate the help). :))

You can find the GitHub repo here:
https://github.com/Kyandroo/personnal-portfolio

Thanks in advance!

**Comment by seyLu** [comment]
#### Hi @Kyandroo 

#### When using static site generators, it is recommended to build and deploy to `gh-pages` branch instead of `main`.
![image](https://github.com/community/community/assets/98249191/5bc713b4-cb1e-4f33-a988-de974e56084c)

#### You can check out these articles on how to deploy Eleventy to Github Pages:
- [How to Deploy Eleventy to Github Pages with Github Actions](https://www.rockyourcode.com/how-to-deploy-eleventy-to-github-pages-with-github-actions/)
- [Making a personal site from start to publish with Eleventy and Github](https://dev.to/ndesmic/making-a-personal-site-from-start-to-publish-with-eleventy-and-github-2ld4)
- [Deploying Eleventy to Github Pages](https://www.justus.ws/tech/deploying-eleventy-to-github-pages/)

**Reply by Kyandroo** [reply]
Hi @seyLu 

Thank you for the help already! I looked a little deeper into the links you sent, but now my homepage is working with css and js, but the navigation to other pages and images done with the templating still won't load in. I also get the following mail from github every time I try to deploy.

![Schermafbeelding 2023-09-14 205717](https://github.com/community/community/assets/56109536/fbafbf9e-1381-4af9-b57a-3f8a836f53b2)

**Reply by Kyandroo** [reply]
This is the monster error I get while deploying:
![Schermafbeelding 2023-09-14 213236](https://github.com/community/community/assets/56109536/666549a7-c23c-4568-95a1-33327ad866c3)

**Reply by seyLu** [reply, marked as answer]
If you're still struggling deploying to github pages, I'd suggest to try and use another platform instead that has better support for eleventy.

Other free static hosting site alternatives include:
- [Cloudfare](https://developers.cloudflare.com/pages/framework-guides/deploy-an-eleventy-site/)
- [Vercel](https://vercel.com/guides/deploying-eleventy-with-vercel)
- [Netlify](https://docs.netlify.com/integrations/frameworks/eleventy/)

**Reply by Kyandroo** [reply]
Thank you very much man! I've used Netlify because I already came across it in my research and it worked very easy.
